### PR TITLE
Updating copyright and notice

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Updating documentation: Copyright, Notice and Compatibility.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VachaShah** [comment]
@opensearch-project/clients Can I please get a review for this?

File: NOTICE.txt [hunk 1]
```diff
@@ -2,5 +2,7 @@ OpenSearch (https://opensearch.org/)
 Copyright OpenSearch Contributors
 
 This product includes software developed by
-Elasticsearch Java Client
-Copyright 2021 Elasticsearch B.V.
+Elasticsearch (http://www.elastic.co).
```
[the remark this answers is not in the document]
Comment (stockholmux) [reply NOTICE.txt:5]: Yeah, that doesn't look right to me.

[the remark this answers is not in the document]
Comment (VijayanB) [reply NOTICE.txt:5]: Right, We should not change existing Notice.txt contents from Elastic.

[the remark this answers is not in the document]
Comment (VachaShah) [reply NOTICE.txt]: Updated it to the original.

**Review by VijayanB — APPROVED** [review]

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**